pipeline/build/azure-pipelines.yaml

```diff
@@ -147,4 +178,89 @@ stages:
           parameters:
             subscription: $(serviceConnection)
             planLocation: $(Pipeline.Workspace)/$(artifactName)/terraform/$(artifactName).tfplan
-            workingDirectory: '$(Pipeline.Workspace)/$(artifactName)/terraform'
\ No newline at end of file
+            workingDirectory: '$(Pipeline.Workspace)/$(artifactName)/terraform'
+
+  ############################################
+  # Apply DSC Configuration. #################
+    - job: Deploy_DSC
+      displayName: Deploy DSC
+      dependsOn: Deploy_Terraform
+      variables:
+        tf_vms: $[ dependencies.Deploy_Terraform.outputs['terraformOutput.tf_vm_names'] ] 
+      steps:
+        - template: /pipeline/templates/deploy-dsc.yaml
+          parameters:
+            serviceConnection: $(serviceConnection)
+            ResourceGroupName: $(ResourceGroupName)
+            automationAccountName: $(ResourceGroupName)
+            patToken: $(patToken) 
+            agentCount: ${{ variables.agentCount }} 
+            devOpsURL: $(devOpsURL)
+            vmNames: $(tf_vms)
+            agentPool: $(agentPool)
+
+  ############################################
+  # Manual Validation of deletion. #######################
+  - ${{ if eq( parameters.env, 'dev' ) }}:
+    - stage: Wait_Destroy
+      dependsOn:
+        - DeployAzureAgent
+      jobs:
+        - job: WaitForApproval
+          displayName: "Wait for approval"
+          pool: server
+          steps:
+            - template: /pipeline/templates/wait.yaml
+
+  # ############################################
+  # # Terraform Destroy. #########################
+  - stage: RemoveDevInfra
```

Comment:
if it runs against staging, these two jobs will not be run.

Reply:
your indentation is wrong then

Wait_Destroy won't be run in staging, but RemoveDevInfra will try and run and then fail as it will then depend on a stage which doesn't exist if running in staging (Wait_Destroy)

Reply:
you are quite right! i could not see the indent for looking! updated :D